web3/_utils/method_formatters.py

```diff
@@ -620,30 +624,45 @@ def subscription_formatter(value: Any) -> Union[HexBytes, HexStr, Dict[str, Any]
         # subscription id from the original subscription request
         return HexStr(value)
 
-    response_key_set = set(value.keys())
+    formatted_value = {"subscription": HexStr(value.get("subscription"))}
+    result = value.get("result")
+    result_key_set = set(result.keys())
 
     # handle dict subscription responses
-    if either_set_is_a_subset(response_key_set, set(BLOCK_FORMATTERS.keys())):
+    if either_set_is_a_subset(
+        result_key_set,
+        set(BLOCK_FORMATTERS.keys()),
+        percentage=90,
```

Comment:
I'm not quite understanding this percentage and how it changes what it was doing previously. Guessing the purpose is to apply specific formatters based on the value passed to `subscription_formatter`. When the value contains most (>90%) of the keys in the formatter type, the formatter is applied? Is it often the case that the value doesnt include all of the same formatters, or is this more to future proof things as more keys are added to the `value`?

Reply:
One of the responses from the `pendingTransaction` subscription uses `yParity` in the place of `v` and we lacked a formatter for `yParity`. In that case it wasn't an exact subset so the formatters for subscription transactions stopped working on those messages entirely. If we are lenient on a percentage, say 90% here, we can distinguish between the different subscription responses and apply formatters appropriately and only those missing a formatter line won't be formatted. Then, if we catch it, all we have to do is update that single formatter and that field too will be formatted appropriately.

This is the [relevant commit](https://github.com/ethereum/web3.py/pull/3096/commits/1107480944730e366df3ccf2300e446960942d0b).